src/Codeception/Util/ReflectionHelper.php

```diff
@@ -121,7 +124,17 @@ public static function getDefaultValue(\ReflectionParameter $param)
             return self::phpEncodeValue($param->getDefaultValue());
         }
 
-        return 'null';
+        $type = $param->getType();
+
+        if (!$type || $type->allowsNull() || !$type->isBuiltin()) {
+            return 'null';
+        }
+
+        $result = null;
+
+        settype($result, $type->getName());
+
+        return var_export($result, true);
```

Comment:
I still down know why does the `phpEncodeValue` method `json_encode` string values.
If there is a good reason for that, it would be better to call `phpEncodeValue` here instead of var_export.

Reply:
I was wondering as well why this is needed? There might be better ways to deal with binary / unicode strings. Don't want to break existing functionality, or introduce security issues, though.

Reply:
Oh, maybe I got this wrong: You say there might be a good reason to call `phpEncodeValue()` instead **because** it uses `json_encode()` for strings?

For this particular use case, the only difference is that `json_encode()`  uses double instead of single quotes (single quotes may be a tiny bit faster in PHP, but probably doesn't matter for empty strings): `""` vs `''`

Generating PHP source using JS encoding generally seems like a bad idea to me.

Reply:
Replaced `var_export()` with a more explicit switch statement. For callbacks and objects, `getDefaultValue()` continues to return "null" as before.